WebCalc/WebCalc/Pages/Index.cshtml.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using WebCalc.Models;
+using WebCalc.Operators;
+
+namespace WebCalc.Pages;
+
+[BindProperties]
+public class Main : PageModel
+{
+    public IOperator[] Operators = { new Divide(), new Minus(), new Multiply(), new Plus() };
+
+    public CalculationData CalculationData { get; set; } = new();
+    public float? Result;
+
+    public void OnPost()
+    {
+        var selectedOperation = Operators.First(op => op.GetSign() == CalculationData.Sign);
```

Comment:
```suggestion
        if (!ModelState.IsValid)
        {
            return Page();
        }
        var selectedOperation = Operators.First(op => op.GetSign() == CalculationData.Sign);
```
И вот тут надо проверять, корректные ли данные нам прилетели с клиента. Иначе нам вполне могут заслать любую строку в качестве символа операции.
(-1)